app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/AlertsViewModel.java

```diff
@@ -1,33 +1,50 @@
 package com.teamagam.gimelgimel.app.mainActivity.viewmodel;
 
+import android.app.Activity;
+
+import com.teamagam.gimelgimel.app.common.launcher.Navigator;
+import com.teamagam.gimelgimel.app.injectors.scopes.PerActivity;
+import com.teamagam.gimelgimel.data.location.LocationFetcher;
 import com.teamagam.gimelgimel.domain.notifications.ConnectivityDisplayer;
 import com.teamagam.gimelgimel.domain.notifications.DisplayDataConnectivityStatusInteractor;
 import com.teamagam.gimelgimel.domain.notifications.DisplayDataConnectivityStatusInteractorFactory;
 import com.teamagam.gimelgimel.domain.notifications.DisplayGpsConnectivityStatusInteractor;
 import com.teamagam.gimelgimel.domain.notifications.DisplayGpsConnectivityStatusInteractorFactory;
 
+import javax.inject.Inject;
+
 /**
  * Manages GPS and Data connectivity status displaying (at the view-model level)
  */
+@PerActivity
 public class AlertsViewModel {
 
     private final DisplayGpsConnectivityStatusInteractorFactory mGpsFactory;
     private final DisplayDataConnectivityStatusInteractorFactory mDataFactory;
-    private final AlertsDisplayer mAlertsDisplayer;
+    private AlertsDisplayer mAlertsDisplayer;
 
     private DisplayGpsConnectivityStatusInteractor mGpsStatusInteractor;
     private DisplayDataConnectivityStatusInteractor mDataStatusInteractor;
 
+    @Inject
+    Navigator mNavigator;
+
+    @Inject
+    LocationFetcher mLocationFetcher;
 
-    public AlertsViewModel(AlertsDisplayer alertsDisplayer,
-                           DisplayGpsConnectivityStatusInteractorFactory gpsFactory,
+    @Inject
+    public AlertsViewModel(DisplayGpsConnectivityStatusInteractorFactory gpsFactory,
                            DisplayDataConnectivityStatusInteractorFactory dataFactory) {
         mGpsFactory = gpsFactory;
         mDataFactory = dataFactory;
+    }
+
+    public void setAlertsDisplayer(AlertsDisplayer alertsDisplayer){
```

Comment:
You don't have to change it, but I'll write it for you to consider again.
If a class cannot operate without a dependency (i.e. the AlertsDisplayer in this case), it should be passed to the class at the constructor. That way, the object cannot be in an illegal state.
This can be done using a factory, as it used to be.